### DISCUSSION TITLE
mosdns开启后，连接数上万，网页卡顿

### DISCUSSION THREAD

**Discussion by buuut** [IrineSistiana/mosdns#232] [Q&A]
开启mosdns时，有一定机率会造成连接数上万，这个时候，上网会有些卡顿，关闭mosdns服务后，肉眼可见的下降，最终稳定在3000左右，不知道有没有遇到同样问题的？

**Comment by IrineSistiana** [comment]
贴配置。

**Reply by buuut** [reply]
log:
  level:                   error
  file:                    ''

plugin:
  - tag:                   main_server
    type:                  server
    args:
      entry:
        - _no_ecs
        - lazy_cache
        - _prefer_ipv4
        - _single_flight
        - main_sequence
      server:
        - protocol:        udp
          addr:            '127.0.0.1:5335'
        - protocol:        tcp
          addr:            '127.0.0.1:5335'

  - tag:                   main_sequence
    type:                  sequence
    args:
      exec:
        - if:
            - query_is_ad_domain
          exec:
            - _block_with_nxdomain
            - _end
        - if:
            - query_is_local_domain
            - '!_query_is_common'
          exec:
            - forward_local
            - _end
        - if:
            - query_is_non_local_domain
          exec:
            - forward_remote
            - _end
        - primary:
            - forward_local
            - if:
                - '!response_has_local_ip'
              exec:
                - _drop_response
          secondary:
            - forward_remote
          fast_fallback:   150
          always_standby:  true

  - tag:                   forward_local
    type:                  fast_forward
    args:
      upstream:
        - addr:            ${{ ./dns0.sh }}
        - addr:            ${{ ./dns1.sh }}


  - tag:                   forward_remote
    type:                  fast_forward
    args:
      upstream:
        - addr:            'tls://8.8.4.4'
          idle_timeout:    60
        - addr:            'tls://101.101.101.101'
          idle_timeout:    20

  - tag:                   lazy_cache
    type:                  cache
    args:
      size:                512000
      lazy_cache_ttl:      259200

  - tag:                   query_is_local_domain
    type:                  query_matcher
    args:
      domain:
        - 'ext:./geosite.dat:cn'

  - tag:                   query_is_non_local_domain
    type:                  query_matcher
    args:
      domain:
        - 'ext:./geosite.dat:geolocation-!cn'

  - tag:                   response_has_local_ip
    type:                  response_matcher
    args:
      ip:
        - 'ext:./geoip.dat:cn'

  - tag: query_is_ad_domain
    type: query_matcher
    args:
      domain:
        - 'ext:./serverlist.txt'

**Comment by buuut** [comment]
#!/bin/bash -e
wanstatus=$(ifconfig | grep wan | wc -l)
if [ "$wanstatus" = 0 ]; then
  echo "223.5.5.5"
  exit 0
fi
DNS0=`ubus call network.interface.wan status | jsonfilter -e '@["dns-server"][0]'`
if [[ $DNS0 =~ ^[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
  echo "$DNS0"
else
  echo "223.5.5.5"
fi
exit 0

**Comment by IrineSistiana** [comment]
定制的 openwrt 路由? 这个环境太复杂我本地没办法复现问题。建议把问题反馈给你用的那个系统的开发者。

**Reply by buuut** [reply]
#!/bin/bash -e
wanstatus=$(ifconfig | grep wan | wc -l)
if [ "$wanstatus" = 0 ]; then
  echo "101.226.4.6"
  exit 0
fi
DNS1=`ubus call network.interface.wan status | jsonfilter -e '@["dns-server"][1]'`
if [[ $DNS1 =~ ^[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
  echo "$DNS1"
else
  echo "101.226.4.6"
fi
exit 0

**Reply by buuut** [reply]
好的，谢谢，已经反馈给QiuSimons。

**Reply by buuut** [reply]
> 定制的 openwrt 路由? 这个环境太复杂我本地没办法复现问题。建议把问题反馈给你用的那个系统的开发者。

理了下使用场景，发现跑京东无线宝pcdn时，打开mosdns插件连接数高峰期可以到12000左右，关闭mosdns后，连接数会掉落到3-4千的样子，不知道是不是无线宝工作原理所致，能否从插件上进行规避此类问题？谢谢

**Reply by yishion819** [reply]
> 好的，谢谢，已经反馈给QiuSimons。

我也遇到这个问题了，请问你是如何解决的？
